examples/flowertune-vit/vitexample/server_app.py

```diff
@@ -1,39 +1,78 @@
 """vitexample: A Flower / PyTorch app with Vision Transformers."""
 
-from logging import INFO
-
 import torch
 from datasets import Dataset, load_dataset
-from flwr.common import Context, ndarrays_to_parameters
-from flwr.common.logger import log
-from flwr.server import ServerApp, ServerAppComponents, ServerConfig
-from flwr.server.strategy import FedAvg
+from flwr.app import ArrayRecord, ConfigRecord, Context, MetricRecord
+from flwr.serverapp import Grid, ServerApp
+from flwr.serverapp.strategy import FedAvg
 from torch.utils.data import DataLoader
 
 from vitexample.task import (
     apply_eval_transforms,
     get_model,
-    get_params,
-    set_params,
     test,
 )
 
 
+# Create ServerApp
+app = ServerApp()
+
+
+@app.main()
+def main(grid: Grid, context: Context) -> None:
+    """Main entry point for the ServerApp."""
+
+    # Define tested for central evaluation
+    dataset_name = context.run_config["dataset-name"]
+    dataset = load_dataset(dataset_name)
+    test_set = dataset["test"]
+    num_rounds = context.run_config["num-server-rounds"]
+
+    # Set initial global model
+    num_classes = context.run_config["num-classes"]
+    model = get_model(num_classes)
+    finetune_layers = model.heads
+    arrays = ArrayRecord(finetune_layers.state_dict())
+
+    # Configure the strategy
+    strategy = FedAvg(
+        fraction_train=0.5,  # Sample 50% of available clients
+        fraction_evaluate=0.0,  # No federated evaluation
+        min_evaluate_nodes=0,
```

Comment:
```suggestion
```

let's also remove this and wait until https://github.com/adap/flower/pull/5867 is merged?